### ISSUE TITLE
separate args from no-unused-vars and introduce no-unused-args

### ISSUE THREAD

**Issue by blockloop** [eslint/eslint#6887]
ESLint Version: v3.2.2

The rule `no-unused-vars` includes the option for unused args in functions. Declaring unused variables is (in my configuration) declared as an error. There isn't a reason to have an unused variable. On the other hand, unused arguments are sometimes necessary for clarity and therefore I like them to be listed as warnings. For example,

```javascript
// Here I like to leave `res` and `next` because I like all http handlers to be declared this way
function getHandler(req, res, next) {
}
```

My first thought was to include the ability to give different levels for args and vars in the `no-unused-vars` rule, but then I immediately thought that maybe the two should be declared separately.

**Comment by vitorbal** [comment]
Thanks for reaching out, @blockloop!
There are already a few args-related options that you could use to achieve this.
For instance, you could use the `argsIgnorePattern` option to whitelist "res" and "next", or you could use the pattern [from the example](http://eslint.org/docs/rules/no-unused-vars#argsignorepattern) so anything starting with an underscore would be ignored:
```json
no-unused-vars: ["error", { "argsIgnorePattern": "^_" }]
```

**Comment by blockloop** [comment]
I'm more concerned with the ability to set different error levels for args and vars.

**Comment by platinumazure** [comment]
@blockloop Unfortunately, as you've observed, ESLint only supports one error level per rule.

I don't know if we're likely to split the rules (I could see an argument for doing so, but at the same time it could be a needless breaking change for the vast majority of our users).

One thing you can do as a workaround is to run ESLint twice with two configurations. The first configuration is your error configuration (including for unused vars), the second is your warning configuration for unused args. Not ideal, but it would get you what you want.

For anything else, you may need to convince the team that your use case is not a corner case and that the pain for other users when separating the rules is worth doing when, as of this writing, it seems only one person is likely to really benefit.

**Comment by nzakas** [comment]
I don't see enough value in splitting the rule in two, as it would be very disruptive for what looks like an extreme edge case. 

Your other option is to create your own custom rule that checks arguments only and set `no-unused-vars` so that it ignores all arguments.

**Comment by alberto** [comment]
Closing as this seems out of scope.
